server/socket_server.go

```diff
@@ -130,15 +130,14 @@ func (s *SocketServer) waitForClose(closeConn chan error, connID int) {
 	if err == io.EOF {
 		s.Logger.Error("Connection was closed by client")
 	} else if err != nil {
-		s.Logger.Error("Connection error", "error", err)
+		s.Logger.Error("Connection error", "err", err)
 	} else {
-		// never happens
-		s.Logger.Error("Connection was closed.")
+		s.Logger.Error("Unexpected error", "err", err)
```

Comment:
don't we have `err == nil` here ?

Reply:
Huh, in deed, I think what happened was that I skimmed over the entire code block and
saw "Connection was closed" being a generic message but didn't notice the `err != nil`. My apologies for wasting time here @ebuchman and @melekes.